src/Interpreters/GraceHashJoin.cpp

```diff
@@ -41,7 +41,7 @@ namespace
     public:
         AccumulatedBlockReader(TemporaryFileStream & reader_,
                                std::mutex & mutex_,
-                               size_t result_block_size_ = DEFAULT_BLOCK_SIZE * 8)
+                               size_t result_block_size_ = 0)
```

Comment:
Don't think that AccumulatedBlockReader makes sense, concatenating blocks can consume extra memory that is not accounted, so disabled it.